File: day01/c/part1.c
#include "helper.h"

#include <stdio.h>
#include <stdlib.h>

int fuel(const int mass) {
    return (mass / 3) - 2;
}

int main()
{
    char* *lines = read_lines("input.txt");

    int total = 0;
    for (int i = 0; i < arrlen(lines); ++i) {
        char* line = lines[i];
        int value = atoi(line);
        total += fuel(value);
    }
    printf("%d\n", total);

    return 0;
}

File: day01/python/part1.py
#!/usr/bin/env python3

import helper


def fuel(mass: int) -> int:
    return (mass // 3) - 2


def main() -> None:
    lines = helper.read_lines("input.txt")
    result = sum(fuel(int(line)) for line in lines)
    print(result)

##############################################################################

if __name__ == "__main__":
    main()
